File: Frontend/src/services/api/matchingService.ts
import createAxiosInstance from './axiosInstance';
import { API_URLS } from '@/config/constants';
import { MatchingScoreResponse } from '@/types';

const matchingApi = createAxiosInstance(API_URLS.AI_MATCHING);

export const matchingService = {
  // Get top matching workers for a job
  findTopMatches: async (jobId: string, limit: number = 10): Promise<MatchingScoreResponse[]> => {
    const response = await matchingApi.get<MatchingScoreResponse[]>(
      `/api/Matching/job/${jobId}?limit=${limit}`
    );
    return response.data;
  },
};

File: Backend/AiMatchingService/src/main/java/com/gramWork/AiMatchingService/Service/MatchingService.java
package com.gramWork.AiMatchingService.Service;

import com.gramWork.AiMatchingService.DTO.MatchingScoreResponse;

import java.util.List;

public interface MatchingService {
    List<MatchingScoreResponse> findTopMatches(String jobId, int limit);
}
